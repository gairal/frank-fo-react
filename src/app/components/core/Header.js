import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import { AppBar, Toolbar, Typography, IconButton, Icon, ListItemIcon, ListItemText, Menu, MenuItem } from '@material-ui/core';

const drawerWidth = 240;
const styles = theme => ({
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('md')]: {
      width: `calc(100% - ${drawerWidth}px)`,
    },
  },
  navIconHide: {
    [theme.breakpoints.up('md')]: {
      display: 'none',
    },
  },
});

class Header extends Component {
  static propTypes = {
    history: PropTypes.shape().isRequired,
    classes: PropTypes.shape().isRequired,
    toggleDrawer: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);

    this.state = {
      anchorEl: null,
      title: '',
    };
  }

  componentDidMount() {
    this.updateTitle(this.props.history.location);
    this.props.history.listen((location) => {
      this.updateTitle(location);
    });
  }

  handleOpenMenu = (event) => {
    this.setState({ anchorEl: event.currentTarget });
  }

  handleRequestClose = () => {
    this.setState({ anchorEl: null });
  }

  updateTitle(location) {
    const title = location.pathname.substr(1) || 'works';

    this.setState(state => ({
      ...state,
      title,
    }));

    document.title = `frank gairal -- ${title}`;
  }

  render() {
    return (
      <AppBar position="fixed" color="primary" className={this.props.classes.appBar}>
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            onClick={this.props.toggleDrawer}
            className={this.props.classes.navIconHide}
          >
            <Icon>menu</Icon>
          </IconButton>
          <Typography variant="display1" component="h1" color="inherit" style={{ flex: 1 }}>
            {this.state.title}
          </Typography>
          <IconButton
            color="inherit"
            aria-label="options"
            onClick={this.handleOpenMenu}
          >
            <Icon>more_vert</Icon>
          </IconButton>
          <Menu
            anchorEl={this.state.anchorEl}
            open={Boolean(this.state.anchorEl)}
            onClose={this.handleRequestClose}
          >
            <a
              href="mailto:frank@gairal.com"
              rel="noopener noreferrer"
              title="frank@gairal.com"
            >
              <MenuItem onClick={this.handleRequestClose}>
                <ListItemIcon>
                  <Icon>email</Icon>
                </ListItemIcon>
                <ListItemText inset primary="frank@gairal.com" />
              </MenuItem>
            </a>
            <a href="tel:+33695676587" rel="noopener noreferrer" title="+33 6 95 67 65 87">
              <MenuItem onClick={this.handleRequestClose}>
                <ListItemIcon>
                  <Icon>phone</Icon>
                </ListItemIcon>
                <ListItemText inset primary="+33 95 67 65 87" />
              </MenuItem>
            </a>
            <a href="https://linkedin.com/in/gairal" target="_blank" rel="noopener noreferrer" title="LinkedIn">
              <MenuItem onClick={this.handleRequestClose}>
                <ListItemIcon>
                  <Icon>link</Icon>
                </ListItemIcon>
                <ListItemText inset primary="LinkedIn" />
              </MenuItem>
            </a>
            <a
              href="/pdf/frank.gairal.pdf"
              target="_blank"
              rel="noopener noreferrer"
              title="resume"
            >
              <MenuItem onClick={this.handleRequestClose}>
                <ListItemIcon>
                  <Icon>file_download</Icon>
                </ListItemIcon>
                <ListItemText inset primary="resume.pdf" />
              </MenuItem>
            </a>
          </Menu>
        </Toolbar>
      </AppBar>
    );
  }
}

const headerWithStyle = withStyles(styles, { withTheme: true })(Header);
const header = withRouter(headerWithStyle);

export default header;
// export default withRouter(withStyles(styles, { withTheme: true })(Header));
