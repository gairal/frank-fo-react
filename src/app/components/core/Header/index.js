import HeaderWith from './Header';

export default HeaderWith;
