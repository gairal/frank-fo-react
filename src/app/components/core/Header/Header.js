import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import { AppBar, Toolbar, Typography, IconButton, Icon } from '@material-ui/core';
import Menu from './HeaderMenu';

const drawerWidth = 240;
const styles = theme => ({
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('md')]: {
      width: `calc(100% - ${drawerWidth}px)`,
    },
  },
  navIconHide: {
    [theme.breakpoints.up('md')]: {
      display: 'none',
    },
  },
});

export class Header extends Component {
  static propTypes = {
    location: PropTypes.shape().isRequired,
    classes: PropTypes.shape().isRequired,
    toggleDrawer: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);

    this.state = {
      title: '',
    };
  }

  componentDidMount() {
    this.updateTitle();
  }

  componentDidUpdate(prevProps) {
    if (this.props.location !== prevProps.location) {
      this.updateTitle();
    }
  }

  updateTitle() {
    const title = this.props.location.pathname.substr(1) || 'works';

    this.setState(state => ({
      ...state,
      title,
    }));

    document.title = `frank gairal -- ${title}`;
  }

  render() {
    return (
      <AppBar position="fixed" color="primary" className={this.props.classes.appBar}>
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            onClick={this.props.toggleDrawer}
            className={this.props.classes.navIconHide}
          >
            <Icon>menu</Icon>
          </IconButton>
          <Typography variant="display1" component="h1" color="inherit" style={{ flex: 1 }}>
            {this.state.title}
          </Typography>
          <Menu />
        </Toolbar>
      </AppBar>
    );
  }
}

export default withRouter(withStyles(styles, { withTheme: true })(Header));
