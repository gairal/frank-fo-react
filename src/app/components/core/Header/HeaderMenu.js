import React, { Component } from 'react';
import { IconButton, Icon, ListItemIcon, ListItemText, Menu, MenuItem } from '@material-ui/core';

export default class HeaderMenu extends Component {
  constructor(props) {
    super(props);

    this.state = { anchorEl: null };
  }

  handleOpenMenu = (event) => {
    this.setState({ anchorEl: event.currentTarget });
  }

  handleRequestClose = () => {
    this.setState({ anchorEl: null });
  }

  render() {
    return (
      <React.Fragment>
        <IconButton
          color="inherit"
          aria-label="options"
          onClick={this.handleOpenMenu}
        >
          <Icon>more_vert</Icon>
        </IconButton>
        <Menu
          anchorEl={this.state.anchorEl}
          open={Boolean(this.state.anchorEl)}
          onClose={this.handleRequestClose}
        >
          <a
            href="mailto:frank@gairal.com"
            rel="noopener noreferrer"
            title="frank@gairal.com"
          >
            <MenuItem onClick={this.handleRequestClose}>
              <ListItemIcon>
                <Icon>email</Icon>
              </ListItemIcon>
              <ListItemText inset primary="frank@gairal.com" />
            </MenuItem>
          </a>
          <a href="tel:+33695676587" rel="noopener noreferrer" title="+33 6 95 67 65 87">
            <MenuItem onClick={this.handleRequestClose}>
              <ListItemIcon>
                <Icon>phone</Icon>
              </ListItemIcon>
              <ListItemText inset primary="+33 95 67 65 87" />
            </MenuItem>
          </a>
          <a href="https://linkedin.com/in/gairal" target="_blank" rel="noopener noreferrer" title="LinkedIn">
            <MenuItem onClick={this.handleRequestClose}>
              <ListItemIcon>
                <Icon>link</Icon>
              </ListItemIcon>
              <ListItemText inset primary="LinkedIn" />
            </MenuItem>
          </a>
          <a
            href="/pdf/frank.gairal.pdf"
            target="_blank"
            rel="noopener noreferrer"
            title="resume"
          >
            <MenuItem onClick={this.handleRequestClose}>
              <ListItemIcon>
                <Icon>file_download</Icon>
              </ListItemIcon>
              <ListItemText inset primary="resume.pdf" />
            </MenuItem>
          </a>
        </Menu>
      </React.Fragment>
    );
  }
}
