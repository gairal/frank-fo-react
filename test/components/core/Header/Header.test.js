/* eslint-disable function-paren-newline,comma-dangle  */
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '@/components/core/Header';

describe('components > core > Header', () => {
  const component = renderer.create(
    <Router>
      <Header
        toggleDrawer={jest.fn}
      />
    </Router>);

  it('Renders correctly default', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Updates it\'s title', () => {
    component.root.instance.history.push('/skills');
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
