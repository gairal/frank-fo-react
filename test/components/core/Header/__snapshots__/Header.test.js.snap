// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > core > Header Renders correctly default 1`] = `
<header
  className="MuiPaper-root-11 MuiPaper-elevation4-17 MuiAppBar-root-3 MuiAppBar-positionFixed-4 MuiAppBar-colorPrimary-9 mui-fixed Header-appBar-1"
>
  <div
    className="MuiToolbar-root-38 MuiToolbar-gutters-39"
  >
    <button
      aria-label="open drawer"
      className="MuiButtonBase-root-46 MuiIconButton-root-40 MuiIconButton-colorInherit-41 Header-navIconHide-2"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-45"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-49"
        >
          menu
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-55"
      />
    </button>
    <h1
      className="MuiTypography-root-62 MuiTypography-display1-66 MuiTypography-colorInherit-81"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      works
    </h1>
    <button
      aria-label="options"
      className="MuiButtonBase-root-46 MuiIconButton-root-40 MuiIconButton-colorInherit-41"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-45"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-49"
        >
          more_vert
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-55"
      />
    </button>
  </div>
</header>
`;

exports[`components > core > Header Updates it's title 1`] = `
<header
  className="MuiPaper-root-11 MuiPaper-elevation4-17 MuiAppBar-root-3 MuiAppBar-positionFixed-4 MuiAppBar-colorPrimary-9 mui-fixed Header-appBar-1"
>
  <div
    className="MuiToolbar-root-38 MuiToolbar-gutters-39"
  >
    <button
      aria-label="open drawer"
      className="MuiButtonBase-root-46 MuiIconButton-root-40 MuiIconButton-colorInherit-41 Header-navIconHide-2"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-45"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-49"
        >
          menu
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-55"
      />
    </button>
    <h1
      className="MuiTypography-root-62 MuiTypography-display1-66 MuiTypography-colorInherit-81"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      skills
    </h1>
    <button
      aria-label="options"
      className="MuiButtonBase-root-46 MuiIconButton-root-40 MuiIconButton-colorInherit-41"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-45"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-49"
        >
          more_vert
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-55"
      />
    </button>
  </div>
</header>
`;
